packages/account/src/providers/provider.ts

```diff
@@ -1864,13 +1863,29 @@ Supported fuel-core version: ${supportedVersion}.`
    * @param nonce - The nonce of the message to retrieve.
    * @returns A promise that resolves to the Message object or null.
    */
-  async getMessageByNonce(nonce: string): Promise<GqlMessage | null> {
-    const { message } = await this.operations.getMessageByNonce({ nonce });
+  async getMessageByNonce(nonce: string): Promise<Message | null> {
+    const { message: rawMessage } = await this.operations.getMessageByNonce({ nonce });
 
-    if (!message) {
+    if (!rawMessage) {
       return null;
     }
 
+    const message: Message = {
```

Comment:
Nit: may be worth sharing this message creation with [above](https://github.com/FuelLabs/fuels-ts/blob/7d74c8c5257d2595325a8acfdb0c5cb4817871ac/packages/account/src/providers/provider.ts#L1662-L1676).